iec61850serversimulator/src/main/java/com/cgi/iec61850serversimulator/Scheduler.java

```diff
@@ -15,98 +16,106 @@
  */
 public class Scheduler {
 
-	private ScheduledThreadPoolExecutor executor;
-	private List<ScheduledFuture<?>> scheduledFutures;
-	private TimeCalculator timeCalculator;
-	private SwitchingMomentCalculator switchingMomentCalculator;
-	private static final Logger logger = LoggerFactory.getLogger(Scheduler.class);
-	private Device device;
+    private ScheduledThreadPoolExecutor executor;
+    private List<ScheduledFuture<?>> scheduledFutures;
+    private TimeCalculator timeCalculator;
+    private SwitchingMomentCalculator switchingMomentCalculator;
+    private static final Logger logger = LoggerFactory.getLogger(Scheduler.class);
+    private Device device;
 
-	public Scheduler(Device device) {
-		ScheduledThreadPoolExecutor executor = (ScheduledThreadPoolExecutor) Executors.newScheduledThreadPool(1);
+    public Scheduler(Device device) {
+        ScheduledThreadPoolExecutor executor = (ScheduledThreadPoolExecutor) Executors.newScheduledThreadPool(1);
 
-		// Should remove cancelled futures, still included .purge when calculating new
-		// switching moments
-		executor.setRemoveOnCancelPolicy(true);
-		this.timeCalculator = new TimeCalculator();
-		device = this.device;
-	}
+        // Should remove cancelled futures, still included .purge when calculating new
+        // switching moments
+        executor.setRemoveOnCancelPolicy(true);
+        this.timeCalculator = new TimeCalculator();
+        this.switchingMomentCalculator = new SwitchingMomentCalculator();
+        this.device = device;
+        this.executor = executor;
 
-	// Base it on feature/example-scheduling code! Triggering at certain
-	// times instead of a continuous check!
+        List<ScheduledFuture<?>> scheduledFutures = new ArrayList<ScheduledFuture<?>>();
+        this.scheduledFutures = scheduledFutures;
```

Comment:
Easier: this.scheduledFutures = new ArrayList<ScheduledFuture<?>>();

Reply:
Done.